scratch_kotlin_lib/src/main/java/dev/sgrem/scratch_kotlin_lib/Week4Homework.kt

```diff
@@ -0,0 +1,160 @@
+package dev.sgrem.scratch_kotlin_lib
+
+fun main() {
+
+    // 1. Task Data:
+    data class Task(
+        val title: String,
+        // var priority: Int = 1,
+        var priority: Priority = Priority.HIGH,
+        var completed: Boolean = false
+    )
+
+    // 2. Task List
+    val taskList: MutableList<Task> = mutableListOf()
+
+    // 3. Add Task:
+    fun addTask(taskList: MutableList<Task>,
+                title: String,
+                // priority: Int = 1,
+                priority: Priority = Priority.HIGH,
+                completed: Boolean = false
+                ){
+        taskList.add(Task(title, priority, completed))
+    }
+
+    addTask(taskList,"Task #10", Priority.LOW, true)
+    addTask(taskList,"Task #20")
+    addTask(taskList,"Task #13", Priority.MEDIUM, true)
+
+    // 4. Sort Tasks:
+    // Simple High-Order function for sorting Tasks by priority only
+    fun sortedTasksHighOrder(
+        taskList: MutableList<Task>,
+        selector: (Task) -> Priority): List<Task> {
+        return taskList.sortedBy {selector(it)}
+    }
```

Comment:
nit: the ending parenthesis can go on a new line.

```suggestion
    fun sortedTasksHighOrder(
        taskList: MutableList<Task>,
        selector: (Task) -> Priority,
    ): List<Task> {
        return taskList.sortedBy {selector(it)}
    }
```